canvas_modules/common-canvas/src/common-properties/ui-conditions/condition-ops/equals.js

```diff
@@ -23,7 +23,10 @@ function op() {
 function evaluate(paramInfo, param2Info, value, controller) {
 	if (paramInfo.control.controlType !== "passwordfield") {
 		const dataType = typeof paramInfo.value;
-		if (typeof param2Info !== "undefined") {
+		if (param2Info) {
+			if (paramInfo.value === null || param2Info.value === null) {
```

Comment:
With this condition defined, it will never get into the `object` case. Should this be moved there instead?

Reply:
I added this here because if the cases where the param2Info.value === null but paramInfo.value=string.  It would drop into the string case and fail.  I removed the check from object since it will never be hit.